doc/reference/reference_lua/pub-sub.rst

```diff
@@ -0,0 +1,172 @@
+.. _pubsub:
+
+Pub/sub system and its events
+=============================
+
+Overview
+--------
+
+In Tarantool, :term:`pub/sub <publisher/subscriber>` is a :term:`notification` system for server-side
+:term:`events <event>`.
+
+..  glossary::
+
+    event
+
+        An event is a state change or a system update update that triggers the action of other systems.
+
+    publisher/subscriber
+
+        A publish/subscribe pattern, also shortened as pub/sub, is a design pattern that consists of
+        event producers (publishers), event consumers (subscribers or watchers), and event channel.
+        Publishers communicate with watchers asynchronously by broadcasting events.
+        They sent events to the pub/sub system.
+        Then the system delivers received events to all the watchers that react to them.
+
+    notification
+
+        A notification is a message created by one side to inform about the occurence of the event and describe
+        it to other components.
+
+The pub/sub pattern is associated with subscriptions.
+Each subscription is defined by the certain key.
+The main feature of the subscriptions is a one-time action.
+It means that if the server generates too many events and a client is too slow,
+the server will not allocate additional memory.
+
+The system processes the following events:
+
+*   ``box.id``
+*   ``box.status``
+*   ``box.election``
+*   ``box.schema``
```

Comment:
I don't see much point in enumerating all system events in an overview - we have a separate section for this and a ToC.